### PR TITLE
Add deprecation classes in importlib.metadata

### PR DESCRIPTION
**Pull request by tungol**
related to https://github.com/python/typeshed/issues/3968

Generally speaking these classes print warnings and then call super().method() without changing anything's type signature, so they don't need much to be filled out here.

### CODE REVIEW

File: stdlib/importlib/metadata/__init__.pyi [hunk 1]
```diff
@@ -69,8 +77,30 @@ class EntryPoint(_EntryPointBase):
     def __hash__(self) -> int: ...
     def __eq__(self, other: object) -> bool: ...
 
-if sys.version_info >= (3, 10):
-    class EntryPoints(list[EntryPoint]):  # use as list is deprecated since 3.10
+if sys.version_info >= (3, 12):
+    class EntryPoints(tuple[EntryPoint]):
```
Comment (AlexWaygood) [inline stdlib/importlib/metadata/__init__.pyi:81]: `tuple[EntryPoint]` means "a `tuple` of length 1, where the sole element is an instance of `EntryPoint`". Is that what you want here, or did you maybe mean

```suggestion
    class EntryPoints(tuple[EntryPoint, ...]):
```

Comment (tungol) [reply stdlib/importlib/metadata/__init__.pyi:81]: The latter. I've also got a stubtest failure, so I'll mark this as draft until I sort that out.

File: stdlib/importlib/metadata/__init__.pyi [hunk 2]
```diff
@@ -68,9 +82,33 @@ class EntryPoint(_EntryPointBase):
 
     def __hash__(self) -> int: ...
     def __eq__(self, other: object) -> bool: ...
+    if sys.version_info >= (3, 11):
+        def __lt__(self, other: object) -> bool: ...
+    if sys.version_info < (3, 12):
+        def __iter__(self) -> Iterator[Any]: ...  # result of iter((str, Self)), really
```
Comment (srittau) [inline stdlib/importlib/metadata/__init__.pyi:88]: So why not use `Iterator[tuple[str, Self]]`?

**Review by srittau — COMMENTED** [review]
Thanks, LGTM, but one question.

File: stdlib/importlib/metadata/__init__.pyi (hunk 2 above)
Comment (tungol) [reply stdlib/importlib/metadata/__init__.pyi:88]: That's an iterator that returns `tuple[str, Self]`. This is an iterator that will always return two items, the first of which is str and the second of which is Self. As far as I know, there's no way to express that in the current system. We could do `Iterator[str | Self]`, but my understanding is that it's better to avoid union return types.

The implementation here is `return iter((self.name, self))`; I meant "iter()` as the runtime function, not a misspelling of the type, but that's a bit confusing because then I switched to the types of the values. Maybe the comment could be better worded, but I'm not sure how without getting too wordy. Might be better to just leave off the comment if it's too confusing. The function was removed in 3.12, so this is going away (eventually) anyway.

**Review by srittau — APPROVED** [review]
